docs/site/migration/auth/passport.md

```diff
@@ -10,8 +10,379 @@ permalink: /doc/en/lb4/migration-auth-passport.html
+      headers: Object.assign({}, this.req.headers),
+    };
+  }
+```
+
+```ts
+  @authenticate('basic')
+  @get('/profiles')
+  async getExternalProfiles(
+    @inject(SecurityBindings.USER) profile: UserProfile,
+  ) {
+    const user = await this.userRepository.findById(
+      parseInt(profile[securityId]),
+      {
+        include: [
+          {
+            relation: 'profiles',
+          },
+        ],
+      },
+    );
+    return user.profiles;
+  }
+```
+
+### Strategy Providers
+
+- Create a provider class that instantiates a passport strategy and wraps it
+  with a strategy adapter. Please read on
+  [@loopback/authentication-passport](https://loopback.io/doc/en/lb4/Authentication-passport.html).
+
+```ts
+/**
+ * basic passport strategy
+ */
+@bind(asAuthStrategy)
+export class BasicStrategy implements AuthenticationStrategy {
+  name = 'basic';
+  passportstrategy: Strategy;
+  strategy: StrategyAdapter<User>;
+
+  constructor(
+    @repository(UserRepository)
+    public userRepository: UserRepository,
+  ) {
+    /**
+     * create a basic passport strategy with verify function to validate credentials
+     */
+    this.passportstrategy = new Strategy(this.verify.bind(this));
+    /**
+     * wrap the passport strategy instance with an adapter to plugin to LoopBack authentication
+     */
+    this.strategy = new StrategyAdapter(
+      this.passportstrategy,
+      this.name,
+      mapProfile.bind(this),
+    );
+  }
+
+  /**
+   * authenticate a request
+   * @param request
+   */
+  async authenticate(request: Request): Promise<UserProfile | RedirectRoute> {
+    return this.strategy.authenticate(request);
+  }
+}
+```
+
+- Passport strategies also require a `verify` function to validate user
+  credentials in the request. Include the `verify` function in the provider
+  class.
+
+```ts
+  /**
+   * authenticate user with provided username and password
+   *
+   * @param username
+   * @param password
+   * @param done
+   *
+   * @returns User model
+   */
+  verify(
+    username: string,
+    password: string,
+    done: (error: any, user?: any) => void,
+  ): void {
+    this.userRepository
+      .find({
+        where: {
+          email: username,
+        }
+      })
+      .then((users: User[]) => {
+        const user = users[0];
+        if (!user.credentials || user.credentials.password !== password) {
+          return done(null, false);
+        }
+        // Authentication passed, return user profile
+        done(null, user);
+      })
+      .catch(err => {
+        done(err);
+      });
+  }
+```
+
+- Bind the strategy provider class to the application
+
+```ts
+export class UserApplication extends BootMixin(
+  ServiceMixin(RepositoryMixin(RestApplication)),
+) {
+  constructor(options: ApplicationConfig = {}) {
+    super(options);
+
+    this.add(createBindingFromClass(BasicStrategy));
+
+    this.sequence(MySequence);
+    this.component(AuthenticationComponent);
+  }
+}
+```
+
+## OAuth2 Strategies
+
```

Comment:
Similar to the above comment, do you mean 
`This section shows how to use .....` ?

Reply:
agreed